### DISCUSSION TITLE
MySQL - not connected (latest release).

### DISCUSSION THREAD

**Discussion by Neu3BecTHo** [Pryaxis/TShock#2627] [Q&A] [answered]
Hello. My problem is that I have made a local web server that includes Apache, PHP, PMA and MySQL. Only now, after entering all the data to connect to a certain database, I get the error shown in the screenshot. ![image](https://user-images.githubusercontent.com/87427271/167288913-0c577525-1b56-4c72-b025-28d189c87fb3.png) I will say that, for example, plugins for minecraft successfully connect to the database, i.e. there were no errors.

**Comment by punchready** [comment]
So when using sqlite instead, it does not throw this error? I doubt this is related to the database because the small amount of code that could throw this has nothing to do with db.

**Reply by Neu3BecTHo** [reply]
Sqlite is used without any errors. In any case, sooner or later I will be able to sort out my problem.

Maybe.

**Comment by Arthri** [comment, marked as answer]
Is this the entire error? is there more that was cut out in the screenshot?

**Reply by Neu3BecTHo** [reply]
![image](https://user-images.githubusercontent.com/87427271/167290849-8d77123c-0b0c-49da-b4ef-4a6bfdf39083.png)
maybe it will help somehow? I doubt that I have configured the database crookedly, but... doubts and in my doubts.

**Reply by Arthri** [reply]
What version of MySQL Server you you using? The connector TShock uses([MySql.Data 6.8.0](https://www.nuget.org/packages/MySql.Data/6.9.8)) is only compatible with MySql Server versions 5.5-5.7

**Reply by punchready** [reply]
Could you translate the russian sentence in there?

**Reply by Neu3BecTHo** [reply]
I understood... Shock uses a version that is much lower than mine. 

If possible, I recommend updating to the latest version of MySQL sometime.

I'll check, of course:>

**Reply by Neu3BecTHo** [reply]
> Could you translate the russian sentence in there?

the specified function is not supported

**Reply by Arthri** [reply]
I think issues #2200 and #2201 could help you

**Reply by punchready** [reply]
Seems like a version issue then.
According to the log, the MySql client tries to open an ssl connection to somewhere, which seems to not be supported by one of the drivers or libraries used.

**Reply by Neu3BecTHo** [reply]
eh... I would like to know more about the reason why MySQL for TShock is like this... outdated?..

**Reply by punchready** [reply]
Because it works for most people, so nobody bothers with putting in the work to update - sad reality of small open source projects

**Reply by Neu3BecTHo** [reply]
Okay, it's time to sort the folders, however.
